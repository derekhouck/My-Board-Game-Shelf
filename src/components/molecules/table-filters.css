.table-filters {
  align-items: center;
  column-gap: 1em;
  display: grid;
  grid-template-columns: repeat(3, minmax(min-content, max-content));
  justify-items: start;
}