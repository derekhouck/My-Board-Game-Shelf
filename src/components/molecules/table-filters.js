import React from 'react';
import { Formik, Field, Form } from 'formik';
import './table-filters.css';

export const TableFilters = props => {
  const { onSubmit } = props;
  return (
    <Formik
      initialValues={{
        name: '',
        status: ''
      }}
      onSubmit={(values, { setSubmitting }) => {
        onSubmit(values);
        setSubmitting(false);
      }}
    >
      {({ isSubmitting }) =>
        <Form
          className="table-filters"
        >
          <div>
            <label htmlFor="name">Name: </label>
            <Field
              id="name"
              name="name"
              type="text"
            />
          </div>
          <div>
            <label htmlFor="status">Status: </label>
            <Field
              as="select"
              id="status"
              name="status"
            >
              <option value="">any</option>
              <option>pending</option>
              <option>approved</option>
              <option>rejected</option>
            </Field>
          </div>
          <button type="submit" disabled={isSubmitting}>
            Filter
          </button>
        </Form>
      }
    </Formik>
  );
};

TableFilters.defaultProps = {
  onSubmit: e => console.log(e)
};