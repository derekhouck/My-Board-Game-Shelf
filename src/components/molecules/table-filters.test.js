import React from 'react';
import { shallow } from 'enzyme';
import { TableFilters } from './table-filters';

describe('<TableFilters />', function () {
  it('renders without crashing', () => {
    const wrapper = shallow(<TableFilters />);
    expect(wrapper).toMatchSnapshot();
  });
});