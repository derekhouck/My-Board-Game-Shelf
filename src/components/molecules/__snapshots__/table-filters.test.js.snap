// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableFilters /> renders without crashing 1`] = `ShallowWrapper {}`;
